src/expr/src/relation/func.rs

```diff
@@ -390,40 +398,69 @@ fn row_number<'a, I>(
     callers_temp_storage: &'a RowArena,
     order_by: &[ColumnOrder],
 ) -> Datum<'a>
+where
+    I: IntoIterator<Item = Datum<'a>>,
+{
+    // We want to use our own temp_storage here, to avoid flooding `callers_temp_storage` with a
+    // large number of new datums. This is because we don't want to make an assumption about
+    // whether the caller creates a new temp_storage between window partitions.
+    let temp_storage = RowArena::new();
+    let datums = row_number_no_list(datums, &temp_storage, order_by);
+
+    callers_temp_storage.make_datum(|packer| {
+        packer.push_list(datums);
+    })
+}
+
+fn row_number_no_list<'a: 'b, 'b, I>(
```

Comment:
Please add doc comments to the functions, here and below. The name is not intuitive enough to know what's going on.

Reply:
Sorry, done